picard.go

```diff
@@ -574,16 +588,35 @@ func getLookupsForDeploy(data interface{}, tableMetadata *tags.TableMetadata, fo
 			}, lookupsToUse...)
 		}
 
-		// Iterate over foreign keys to check in reverse so we can remove items without consequence
+		// Iterate over foreign keys to check in reverse so we can remove foreign keys that already have values
 		for i := len(foreignKeysToCheck) - 1; i >= 0; i-- {
 			foreignKeyToCheck := foreignKeysToCheck[i]
 			fkValue := item.FieldByName(foreignKeyToCheck.FieldName)
-			if fkValue.IsValid() && fkValue.String() != "" && foreignKeyToCheck.NeedsLookup {
+			fkValueType := fkValue.Kind()
+			fkValueActual := ""
+			if fkValueType == reflect.String {
+				fkValueActual = fkValue.String()
+			} else if fkValueType == reflect.Pointer {
+				v := fkValue.Elem()
+				if v.IsValid() {
+					fkValueActual = v.String()
+				}
+			} else {
+				fkValueActual = fkValue.String()
+			}
+			if fkValue.IsValid() && fkValueActual != "" && foreignKeyToCheck.NeedsLookup {
+				// Take this foreign key out because we already have its id and do not need to look it up
 				foreignKeysToCheck = append(foreignKeysToCheck[:i], foreignKeysToCheck[i+1:]...)
 				lookupsToUse = append(lookupsToUse, tags.Lookup{
 					MatchDBColumn:       foreignKeyToCheck.KeyColumn,
 					MatchObjectProperty: foreignKeyToCheck.FieldName,
 				})
+			} else {
+				// We don't have the id value for this foreign key so it does need a lookup
+				// But we should only pass on the ones where the original data has the values to lookup
+				if !hasForeignKeyData(item, foreignKeyToCheck) {
```

Comment:
This is to account for the fact that HeaderValues and ParameterValues have a RestObjectID and an ActionID field, and both are marked as lookup fields, but only one will ever be filled out. The one that doesn't have a value should be skipped